torch/_dynamo/variables/lists.py

```diff
@@ -489,7 +486,12 @@ def python_type(self):
         return slice
 
     def as_python_constant(self):
-        return slice(*[x.as_python_constant() for x in self.items])
+        def convert(x):
+            if isinstance(x, variables.SymNodeVariable):
+                return x.sym_num
```

Comment:
Are we allowed to return SymInt for `as_python_constant`? Seems questionable: for example SymNodeVariable does not implement `as_python_constant`. This is a big NO from me.

Reply:
I don't remember why I did this, this is fishy. Let me check.

Reply:
Ah, I recall. I added a comment. This is an unfortunate intermediary state. 

I don't mind actually doing the legwork refactor here.

Reply:
This still needs fixing?

Reply:
I added a comment, I think this is okay to land as is, but if you disagree, I am happy to do that legwork here. LMK what you think.